shell/platform/darwin/graphics/FlutterDarwinExternalTextureMetal.mm

```diff
@@ -0,0 +1,232 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/graphics/FlutterDarwinExternalTextureMetal.h"
+
+#include "flutter/fml/logging.h"
+#include "third_party/skia/include/core/SkYUVAInfo.h"
+#include "third_party/skia/include/gpu/GrBackendSurface.h"
+#include "third_party/skia/include/gpu/GrDirectContext.h"
+#include "third_party/skia/include/gpu/GrYUVABackendTextures.h"
+#include "third_party/skia/include/gpu/mtl/GrMtlTypes.h"
+
+@implementation FlutterDarwinExternalTextureMetal {
+  CVMetalTextureCacheRef _textureCache;
+  NSObject<FlutterTexture>* _externalTexture;
+  BOOL _textureFrameAvailable;
+  sk_sp<SkImage> _externalImage;
+  CVPixelBufferRef _lastPixelBuffer;
+  OSType _pixelFormat;
+}
+
+- (instancetype)initWithTextureCache:(nonnull CVMetalTextureCacheRef)textureCache
+                           textureID:(int64_t)textureID
+                             texture:(NSObject<FlutterTexture>*)texture {
+  if (self = [super init]) {
+    _textureCache = textureCache;
```

Comment:
Isn't this missing a `CFRetain` and subsequent release in the dealloc? We might end up with a dangling reference otherwise.

Reply:
Done!